### ISSUE TITLE
accessors should reject ether

### ISSUE THREAD

**Issue by chriseth** [ethereum/solidity#602]
Accessor functions should reject ether sent to them.

**Comment by LefterisJP** [comment]
Thanks for this. It's a much needed change. For now the only way around this is to do something like this: https://github.com/slockit/DAO/pull/185/files#diff-00f5402e5918b82e1560a8173c0aa689R99

**Comment by chriseth** [comment]
Would it be better to just allow modifiers for variables?

**Comment by VoR0220** [comment]
do explain this modifier for variables idea

**Comment by chriseth** [comment]
Modifiers can only be added to functions, but we could add them in the same way to a state variable which means it will be applied to its accessor function. A modifier that would solve exactly this issue is the one named after Emmy Noether:

```
modifier noether { if (msg.value > 0) throw; _ }
```

Example of usage:

```
contract c {
  uint public noether x;
}
```

**Comment by VoR0220** [comment]
seems reasonable only if they are state variables. Particularly when thought of in the context of public getters.

**Comment by chriseth** [comment]
Fixed by #962
